harp/reader.py

```diff
@@ -51,13 +51,21 @@ class DeviceReader:
     def __init__(self, device: Model, registers: dict[str, RegisterReader]) -> None:
```

Comment:
The signature here would probably have to be changed into `dict[Union[str, int], RegisterReader]`.

Reply:
Actually, on second thought, no. For the constructor passing this dict is enough, we then construct the multi-key dictionary from this one.